### ISSUE TITLE
RAT proof feature: Counter counts wrong if test is registered when the test already returned negative

### ISSUE THREAD

**Issue by Ein-Tim** [corona-warn-app/cwa-app-android#3559]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Samsung Galaxy S10+
- Android version: 11
- App version: 2.3.4

## Describe the bug
If a user takes a RAT on the 01.06. at 12h but only registers the test result in the Corona-Warn-App on the 02.06. at 12h, [the counter](https://www.coronawarn.app/assets/screenshots/2-4/de/android/rat_7.png), which is shown under Android when a RAT returns negative, does not count the hours since the test result is available (and correctly shows that the test result is available since 24h in the example) but counts the time since the user has registered the test in the app.

## Steps to reproduce the issue

1. Take a RAT at a test site supporting the CWA ([this](https://map.schnelltestportal.de) map helps to find one)
2. Register the test in the CWA one day after the swap was taken and the test result was avaliable
3. See that the counter is wrong, as described above. 

## Expected behaviour

The counter should count the hh:mm:ss since the test result is available, but not the hh:mm:ss since the test result was registered in the app.

## Additional context

This bug report is based on the information a user shared on Twitter: https://twitter.com/Fridde82/status/1408325023429976068?s=20

**Comment by heinezen** [comment]
@Ein-Tim 

Thank you for reporting. The dev team will investigate it. 

----
Corona-Warn-App Open Source Team

**Comment by heinezen** [comment]
@Ein-Tim 

Dev team says they cannot reproduce the issue on 2.3 and 2.5. Do you have more information about the circumstances of the issue? Either way, we're trying to eliminate possible causes in 2.5.

----
Corona-Warn-App Open Source Team

**Comment by Ein-Tim** [comment]
@heinezen 

Thank you for the information.
What information do you need, I could ask the Twitter user once more.

**Comment by heinezen** [comment]
@Ein-Tim 

Can you ask for the scenario that took place? I.e. the place where they made the test, when exactly (time of day, date) the test was made, when exactly it was scanned by the user and whether the displayed timer changed later. Anything about the process that they can think of.

They also don't need to send it to you, the can use our mail address: coronawarnapp@rki.de , but must mention the Github issue and Jira ticket ID so that we can associate the report.

Thanks for assisting!

----
Corona-Warn-App Open Source Team

**Comment by Ein-Tim** [comment]
@heinezen 

I'll ask them and also mention the E-Mail Adress! I'll keep you posted.

**Comment by RaoulDuke25** [comment]
Test was made on 2. July 2021, 18:22, Schnelltest Service Hamburg, testflow.eu, station Aspria Uhlenhorst. Was imported into CWA on 2. July, 21:36. Counter started from zero. 
![Screenshot_20210704_130944](https://user-images.githubusercontent.com/1859656/124382992-7aff7a80-dcca-11eb-86de-a65ace578090.jpg)
![Uploading Screenshot_20210704_132039.jpg…]()

**Comment by heinezen** [comment]
@RaoulDuke25 Thank you!

That should help a lot. I've forwarded the info to the dev team.

----
Corona-Warn-App Open Source Team

**Comment by mtwalli** [comment]
@RaoulDuke25  Thank you very much for your report , In order to be able to find out  what went wrong in your side, it would be great if you can provide us with a bug report https://developer.android.com/studio/debug/bug-report
or using the error report feature in the app if it did capture the logs already ,this can be accessed by:
`Three dots In the main screen -> App-Informationen -> Fehlerberichte`

**Comment by vaubaehn** [comment]
@mtwalli 
In case you can't reproduce the issue with CWA, maybe some additional hints can help:
There are chances for a bug in quicktest-backend, test-result-server, or even 3rd-party implementations for test result transmission.
Some related issues/PRs:
Implementation of sc in test-result-server:
Issue: https://github.com/corona-warn-app/cwa-testresult-server/issues/93
PR: https://github.com/corona-warn-app/cwa-testresult-server/pull/94
There once was a report that setting sc was not working like expected: https://github.com/corona-warn-app/cwa-quick-test-backend/issues/115

Implementation of sc in quicktest-backend:
Issue: https://github.com/corona-warn-app/cwa-quick-test-backend/issues/82
PR: https://github.com/corona-warn-app/cwa-quick-test-backend/pull/98
Additional fix to update sc: 
https://github.com/corona-warn-app/cwa-quick-test-backend/pull/101

Or has a dcc sc timestamp mix up been introduced? https://github.com/corona-warn-app/cwa-quick-test-backend/pull/129

Related to this issue here: https://github.com/corona-warn-app/cwa-documentation/issues/630

So, if you can't find anything on CWA side, maybe @mschulte-tsi and @ascheibal can assist.

**Comment by mtwalli** [comment]
@vaubaehn  Thanks for the detailed response, Unfortunately I was not able to identify the issue in the client side, We rely on the `sc` time to start the timer , if not available we take the timestamp from the Qr-Code payload istself if valid , if it is not valid we throw an error. So there is no way to use the time when test was actually scanned in CWA, and I have tested it using CWA environment , Timer displayed the right time. I would assume that the `sc` for some test labs is not accurate, maybe someone from the sever side can assist here to find out ?

**Comment by vaubaehn** [comment]
Hi @mtwalli , from information I was able to gather in related repos, it could be indeed that a bad design of implementation on 3rd-party-side may cause these issues. I created an issue in quick-test-onboarding about wrong information to partners that 'sc' can be transmitted optional. I asked to contact CWA team about the sc issues, and I assume you'll hear soon from them.

**Comment by vaubaehn** [comment]
@mschulte-tsi @ascheibal @mtwalli 
I have some new/interesting information after I tested one example flow myself.

Testcenter: dm
3rd-party-provider: NETSYNO
DCC support: currently not, but planned

Test appointment taken: 09.07.2021 10:00 for testing same day at 12:00
Test/sample collected (SC): 09.07.2021 12:04
Confirmation received that test result is available: 09.07.2021 12:21

Due to the new TestVerordnung, it needs a confirmation by the testee that the test actually had been carried out. In case of NETSYNO it is done by clicking on a confirmation link that is included in the 'confirmation E-Mail' (received 12:21).

Confirmed that test was carried out: 09.07.2021 14:20
Received 2 PDFs in signed E-Mails (👍 ) from NETSYNO's backend, one with Certificate, one with Password: 09.07.2021 14:20

Scanned CWA-QR code received at Appointment Time to register test in CWA: 09.07.2021 15:03

CWA displays: Test result available since 42 minutes, issued at 09.07.2021 14:20
<img src="https://user-images.githubusercontent.com/68376758/125085121-87b90f80-e0ca-11eb-9970-a7c6a6cad4d4.png" width="200">

This means:
- sc timestamp is in this case the time when user confirms that test was carried out
- user can influence this time and by this influence the time from what time on the test result will be valid
- if the flow won't be corrected, DCC validity can be manipulated as soon as NETSYNO supports DCCs.
- it's unclear, whether NETSYNO's backend provided a wrong sc timestamp, or no timestamp and the timestamp was set by TSI's test-result-server accordingly wrong.

I can proof my test registration time, as I created an error report and uploaded it:

2021.07.09 - 15:05:03
ID CC85089433434C1FE52F 

I'll inform NETSYNO shortly in parallel about this issue.

Beside this: I really like NETSYNO's flow personally! 👍 

This issue does not affect NETSYNO alone, but potentially all 3rd-party partners, so it will need a bigger effort to correct this.

**Comment by dmr** [comment]
Thank you pointing out the details here. I feel like the change was introduced by 1.7. changes. Before that date the CWA showed the date when the test was announced.

We adjusted our backend to include the correct date.
The test result time is a new field and our implementation just left it out as it didn't exist earlier.

**Comment by vaubaehn** [comment]
Hi @dmr ,
thanks for your fast fix! I provided some suggestions towards TSI on how to continue with this issue regarding other 3rd parties: https://github.com/corona-warn-app/cwa-quicktest-onboarding/issues/24#issuecomment-878084791

Thanks for your feedback, and good luck with implementing DCCs!

**Comment by Ein-Tim** [comment]
Please transfer this issue to the documentation repository, since iOS is [also affected here](https://twitter.com/crowslegend/status/1417018071827554306?s=21).

**Comment by vaubaehn** [comment]
@Ein-Tim could you maybe ask the user which testcenter/testflow provider he used?

@dsarkar and @heinezen as this issue apparently needs to be fixed by TSI,  could you please remove the "Fix 2.6 label"? Is there any progress in this issue that you can share here?
Thank you

**Comment by Ein-Tim** [comment]
@vaubaehn I asked him, but sadly I didn't get an answer...
